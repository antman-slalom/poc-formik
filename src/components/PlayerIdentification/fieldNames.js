const NAME = 'name';
const NUMBER = 'number';

export default {
    NAME,
    NUMBER,
};
