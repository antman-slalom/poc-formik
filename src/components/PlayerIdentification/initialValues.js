import fieldsNames from './fieldNames';

export default {
    [fieldsNames.NAME]: '',
    [fieldsNames.NUMBER]: '',
};
