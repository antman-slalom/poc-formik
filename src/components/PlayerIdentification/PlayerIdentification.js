import React from 'react';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';

import TextField from '../../formik/TextField';
import isEqual from 'react-fast-compare';
import { withNamespace } from '../../formik/utils';

import fieldNames from './fieldNames';

/*
    TODO - things to talk to Will about
    1. import withNamespace
    2. where to put the fomrik shared components
    3. unit testing the formik shared components
    4. what peices should be brought over now, into master
*/

class PlayerIdentification extends React.Component {
    shouldComponentUpdate(nextProps) {
        return !isEqual(nextProps, this.props);
    }

    render() {
        const {
            classes,
            namespace,
        } = this.props;

        return (
            <React.Fragment>
                <Typography variant='h5'>Player Identification: </Typography>
                <Grid className={ classes.Grid } container>
                    <Grid className={ classes.GridItem } item xs={ 10 }>
                        <TextField
                            label='Name'
                            name={ withNamespace(namespace, fieldNames.NAME) }
                            variant='outlined'
                        />
                    </Grid>
                    <Grid className={ classes.GridItem } item xs={ 2 }>
                        <TextField
                            label='Number'
                            name={ withNamespace(namespace, fieldNames.NUMBER) }
                            variant='outlined'
                        />
                    </Grid>
                </Grid>
            </React.Fragment>
        );
    }
}

export default PlayerIdentification;
