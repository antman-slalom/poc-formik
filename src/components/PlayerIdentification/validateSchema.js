import * as yup from 'yup';
import fieldNames from './fieldNames';

const MAX_NAME_LENGTH = 20;

const REQUIRED_MESSAGE = 'Required';
const NAME_MESSAGE = `The middle initial can be no more than ${ MAX_NAME_LENGTH } characters long`;

export default yup.object({
    [fieldNames.NAME]: yup
        .string()
        .required(REQUIRED_MESSAGE)
        .max(MAX_NAME_LENGTH, NAME_MESSAGE),
    [fieldNames.NUMBER]: yup
        .string()
        .required(REQUIRED_MESSAGE)
        .max(MAX_NAME_LENGTH, NAME_MESSAGE),
});
