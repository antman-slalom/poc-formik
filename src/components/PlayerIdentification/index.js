import { withStyles } from '@material-ui/styles';
import PlayerIdentification from './PlayerIdentification';

import styles from './styles';

export default withStyles(styles)(PlayerIdentification);
