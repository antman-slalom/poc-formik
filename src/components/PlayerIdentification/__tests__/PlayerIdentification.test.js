import React from 'react';
import { shallow } from 'enzyme';

import PlayerIdentification from '../PlayerIdentification';

describe('PlayerIdentification', () => {
    let wrapper;
    let instance;

    const props = {
        classes: {},
        namespace: 'TEST',
    };

    beforeEach(() => {
        wrapper = shallow(<PlayerIdentification { ...props } />);
        instance = wrapper.instance();
    });

    afterEach(() => {
        // clears all mocks to prevent mock bleeding from test to test,
        jest.clearAllMocks();

        // mock the API service after the clear
        jest.mock('../../../services/AdminService');
    });

    // =================================
    // Render and Snapshot
    // =================================

    it('matches snapshot', () => {
        expect(wrapper.instance()).toMatchSnapshot();
    });

    it('should render', () => {
        expect(wrapper.instance()).not.toBeNull();
    });

    // =================================
    // Components call correct functions
    // =================================

    // N/A

    // =================================
    // Functions act as expected
    // =================================
    it('Test shouldComponentUpdate to be false', () => {
        const result = instance.shouldComponentUpdate(props);
        expect(result).toBe(false);
    });

    it('Test shouldComponentUpdate to be true', () => {
        const result = instance.shouldComponentUpdate({});
        expect(result).toBe(true);
    });
});
