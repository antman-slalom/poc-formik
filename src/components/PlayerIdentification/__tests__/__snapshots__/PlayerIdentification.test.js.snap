// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerIdentification matches snapshot 1`] = `
PlayerIdentification {
  "context": Object {},
  "props": Object {
    "classes": Object {},
    "namespace": "TEST",
  },
  "refs": Object {},
  "setState": [Function],
  "state": null,
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useResponder": [Function],
        "useState": [Function],
      },
      "_element": <PlayerIdentification
        classes={Object {}}
        namespace="TEST"
      />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_renderPhaseUpdates": null,
      "_rendered": <React.Fragment>
        <ForwardRef(WithStyles)
          variant="h5"
        >
          Player Identification: 
        </ForwardRef(WithStyles)>
        <ForwardRef(WithStyles)
          container={true}
        >
          <ForwardRef(WithStyles)
            item={true}
            xs={10}
          >
            <TextField
              label="Name"
              name="TEST.name"
              variant="outlined"
            />
          </ForwardRef(WithStyles)>
          <ForwardRef(WithStyles)
            item={true}
            xs={2}
          >
            <TextField
              label="Number"
              name="TEST.number"
              variant="outlined"
            />
          </ForwardRef(WithStyles)>
        </ForwardRef(WithStyles)>
      </React.Fragment>,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
}
`;
